src/Router.ts

```diff
@@ -0,0 +1,198 @@
+import { after } from 'dojo-core/aspect';
+import Evented from 'dojo-core/Evented';
+import { Handle, EventObject } from 'dojo-core/interfaces';
+import { assign } from 'dojo-core/lang';
+import Promise from 'dojo-core/Promise';
+import CancelNavigationError from './errors/CancelNavigationError';
+import MissingRouteError from './errors/MissingRouteError';
+import has from './has';
+import HashSource from './HashSource';
+import HtmlHistorySource from './HtmlHistorySource';
+import { CancelableNavigationArgs, NavigationArgs, RouterArgs, RouterSource } from './interfaces';
+import { joinPath, normalizePath } from './PathRule';
+import RouteGroup from './RouteGroup';
+
+const DefaultSource = has('html5-history') ? HtmlHistorySource : HashSource;
+let defaultSource: RouterSource = null;
+
+if (has('host-browser')) {
+	defaultSource = new (<any> DefaultSource)();
+}
```

Comment:
Oh, you're not asking whether the use of `const` for `DefaultSource` is valid; you're suggesting that the same could be done for `defaultSource`. Gotcha.